dell/src/main/java/org/apache/iceberg/dell/EcsSeekableInputStream.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.dell;
+
+import com.emc.object.Range;
+import com.emc.object.s3.S3Client;
+import java.io.IOException;
+import java.io.InputStream;
+import org.apache.iceberg.io.SeekableInputStream;
+
+/**
+ * A {@link SeekableInputStream} impl that warp {@link S3Client#readObjectStream(String, String, Range)}
+ * <p>
+ * 1. The stream is only be loaded when start reading.
+ * <p>
+ * 2. This class won't cache any bytes of content. It only maintains pos of {@link SeekableInputStream}
+ * <p>
+ * 3. This class is not thread-safe.
+ */
+class EcsSeekableInputStream extends SeekableInputStream {
+
+  private final S3Client client;
+  private final EcsURI uri;
+
+  /**
+   * Mutable pos set by {@link #seek(long)}
+   */
+  private long newPos = 0;
+  /**
+   * Current pos of object content
+   */
+  private long pos = -1;
+  private InputStream internal;
```

Comment:
nit: name this a a noun, such as `stream` or `delegate`

Reply:
Done. Changed it to `internalStream`.